triplea-game/triplea
# Support macOS integration on Java 9 or later

## Overview

Fixes #3297.

Implements the pure reflection proposal discussed in that issue (i.e. option (3) in https://github.com/triplea-game/triplea/issues/3297#issuecomment-374384970).

## Functional Changes

None.

## Refactoring Changes

I consolidated all the macOS integration stuff in a single class (`MacOsIntegration`) rather than having it split across three classes.

## Manual Testing Performed

None because I'm unable to test on macOS.  After Christmas, I plan on soliciting help from the community via the forum to verify these changes are working properly on both Java 8 and Java 11.

However, even if bugs are reported, I don't expect much of the code to change from a high-level perspective.  Therefore, please feel free to review this before the **&lt;TESTING>** label is removed.
